system_setup/models/wslconfadvanced.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2015 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import subprocess
+import attr
+
+from subiquitycore.utils import run_command
+
+log = logging.getLogger('subiquity.models.wsl_configuration_advanced')
+
+# TODO WSL: Remove all common attributes with wslconfbase
+
+
+@attr.s
+class WSLConfigurationAdvanced(object):
+    gui_theme = attr.ib()
+    gui_followwintheme = attr.ib()
+    legacy_gui = attr.ib()
+    legacy_audio = attr.ib()
+    adv_ip_detect = attr.ib()
+    wsl_motd_news = attr.ib()
+    automount = attr.ib()
+    mountfstab = attr.ib()
+    custom_path = attr.ib()
+    custom_mount_opt = attr.ib()
+    gen_host = attr.ib()
+    gen_resolvconf = attr.ib()
+    interop_enabled = attr.ib()
+    interop_appendwindowspath = attr.ib()
+
+
+class WSLConfigurationAdvancedModel(object):
+    """ Model representing integration
+    """
+
+    def __init__(self):
+        self._wslconfadvanced = None
+        # TODO WSL: Load settings from system
+
+    def apply_settings(self, result, is_dry_run=False):
+        d = {}
+        # TODO: placholder settings; should be dynamically assgined using
+        # ubuntu-wsl-integration
+        d['custom_path'] = result.custom_path
+        d['custom_mount_opt'] = result.custom_mount_opt
+        d['gen_host'] = result.gen_host
+        d['gen_resolvconf'] = result.gen_resolvconf
+        d['interop_enabled'] = result.interop_enabled
+        d['interop_appendwindowspath'] = result.interop_appendwindowspath
+        d['gui_theme'] = result.gui_theme
+        d['gui_followwintheme'] = result.gui_followwintheme
+        d['legacy_gui'] = result.legacy_gui
+        d['legacy_audio'] = result.legacy_audio
+        d['adv_ip_detect'] = result.adv_ip_detect
+        d['wsl_motd_news'] = result.wsl_motd_news
+        d['automount'] = result.automount
+        d['mountfstab'] = result.mountfstab
+        self._wslconfadvanced = WSLConfigurationAdvancedModel(**d)
```

Comment:
is there a typo here? shouldn't it create a `WSLConfigurationAdvanced` instance?

```
Traceback (most recent call last):
  File "/home/jpnurmi/Projects/canonical/ubuntu-desktop-installer/packages/subiquity_client/subiquity/subiquity/common/api/server.py", line 123, in handler
    result = await implementation(**args)
  File "/home/jpnurmi/Projects/canonical/ubuntu-desktop-installer/packages/subiquity_client/subiquity/system_setup/server/controllers/wslconfadvanced.py", line 189, in POST
    self.model.apply_settings(data, self.opts.dry_run)
  File "/home/jpnurmi/Projects/canonical/ubuntu-desktop-installer/packages/subiquity_client/subiquity/system_setup/models/wslconfadvanced.py", line 71, in apply_settings
    self._wslconfadvanced = WSLConfigurationAdvancedModel(**d)
TypeError: __init__() got an unexpected keyword argument 'custom_path'
```

Reply:
Are you in first run mode or reconfiguration mode? POST in reconfigure is not fully implemented and should be fixed by the work Patrick is doing.

Reply:
I'm running some simple POST/GET tests in first run mode just to verify that the data structure is in the expected format.